verifier.sh

```diff
@@ -0,0 +1,686 @@
+
+NL="
+"
+TB="	"
+
+#
+#  remote init files
+cat >.gem_init.sh <<EOF
+export GEM_SET_DEBUG=$GEM_SET_DEBUG
+set -e
+if [ -n "\$GEM_SET_DEBUG" -a "\$GEM_SET_DEBUG" != "false" ]; then
+    export PS4='+\${BASH_SOURCE}:\${LINENO}:\${FUNCNAME[0]}: '
+    set -x
+fi
+source .gem_ffox_init.sh
+EOF
+
+cat >.gem_ffox_init.sh <<EOF
+export GEM_FIREFOX_ON_HOLD=$GEM_FIREFOX_ON_HOLD
+if [ "\$GEM_FIREFOX_ON_HOLD" ]; then
+    sudo apt-mark hold firefox firefox-locale-en
+else
+    sudo apt-get update
+    ffox_pol="\$(apt-cache policy firefox)"
+    ffox_cur="\$(echo "\$ffox_pol" | grep '^  Installed:' | sed 's/.*: //g')"
+    ffox_can="\$(echo "\$ffox_pol" | grep '^  Candidate:' | sed 's/.*: //g')"
+
+    if [ "\$ffox_cur" != "\$ffox_can" ]; then
+        echo "WARNING: firefox has been upgraded, run it to accomplish update operations"
+        sudo apt-get -y upgrade
+        sudo apt-get -y install wmctrl
+        export DISPLAY=:1
+        firefox &
+        ffox_pid=\$!
+        st="none"
+        for i in \$(seq 1 1000) ; do
+            if ! wmctrl -l >/dev/null; then
+                sleep 0.1
+                continue
+            fi
+            ffox_wins="\$(export DISPLAY=:1 ; wmctrl -l | grep -i "firefox" || true)"
+            if [ "\$st" = "none" ]; then
+                if echo "\$ffox_wins" | grep -qi 'update'; then
+                    st="update"
+                elif echo "\$ffox_wins" | grep -qi 'mozilla'; then
+                    break
+                fi
+            elif [ "\$st" = "update" ]; then
+                if echo "\$ffox_wins" | grep -qvi 'update'; then
+                    break
+                fi
+            fi
+            sleep 0.02
+        done
+        kill \$ffox_pid || true
+        sleep 2
+    fi
+fi
+EOF
+
+#
+#  functions
+copy_common () {
+    scp "${lxc_ip}:ssh.log" "out/${1}_ssh_history.log" || true
+    scp "${lxc_ip}:.pip/pip.log" "out/${1}_pip_history.log" || true
+}
+
+copy_dev () {
+    scp "${lxc_ip}:$GEM_GIT_PACKAGE/xunit-platform-dev.xml" "out/" || true
+    scp "${lxc_ip}:$GEM_GIT_PACKAGE/dev_*.png" "out/" || true
+    scp "${lxc_ip}:$GEM_GIT_PACKAGE/runserver.log" "out/dev_runserver.log" || true
+}
+
+copy_prod () {
+    scp "${lxc_ip}:/var/log/apache2/access.log" "out/prod_apache2_access.log" || true
+    scp "${lxc_ip}:/var/log/apache2/error.log" "out/prod_apache2_error.log" || true
+    scp "${lxc_ip}:/var/log/tomcat7/catalina.out" "out/prod_tomcat7_catalina.log" || true
+    scp "${lxc_ip}:$GEM_GIT_PACKAGE/openquakeplatform/xunit-platform-prod.xml" "out/" || true
+    scp "${lxc_ip}:$GEM_GIT_PACKAGE/openquakeplatform/prod_*.png" "out/" || true
+    scp "${lxc_ip}:/etc/openquake/platform/local_settings.py" "out/prod_local_settings.py" || true
+}
+
+#
+#  sig_hand - manages cleanup if the build is aborted
+#
+sig_hand () {
+    trap ERR
+    echo "signal trapped"
+    if [ "$lxc_name" != "" ]; then
+        set +e
+        ssh -t  $lxc_ip "cd ~/$GEM_GIT_PACKAGE; . platform-env/bin/activate"
```

Comment:
This activates a virtual environment in a session that dies immediately.  It looks useless.